src/script/lua_api/l_env.cpp

```diff
@@ -651,38 +651,49 @@ int ModApiEnvMod::l_find_nodes_in_area(lua_State *L)
 	INodeDefManager *ndef = getServer(L)->ndef();
 	v3s16 minp = read_v3s16(L, 1);
 	v3s16 maxp = read_v3s16(L, 2);
+	v3s16 cube = maxp - minp + 1;
+
+	/* Limit for too large areas, assume default values
+	 * and give tolerances of 1 node on each side
+	 * (chunksize * MAP_BLOCKSIZE + 2)^3 = 551368
+	*/
+	if (cube.X * cube.Y * cube.Z > 551368) {
```

Comment:
are you sure cube.X * cube.Y * cube.Z cannot overflow ? s16 * s16 * s16 = s16 => overflow, in think it's better to have a bigger integer ton contain this result